app/src/main/java/com/example/nicocalculator/mvp/presenter/Presenter.kt

```diff
@@ -0,0 +1,117 @@
+
+class Presenter(private val model: CalculatorModel, private val view: KeyView) {
+
+    companion object {
+        private const val VALUE_EMPTY_TEXT = ""
+        private const val VALUE_ZERO = 0
+        private const val DECIMAL_SEPARATOR = '.'
+        private const val VALUE_MIN_CAR = 48
+        private const val VALUE_MAX_CHAR = 57
+    }
+
+    fun onDeleteButtonPressed() {
+        var textOperations = model.getTextOperations()
+        textOperations =
+            if (textOperations.isNotEmpty()) textOperations.subSequence(0 until textOperations.length - 1) as String
+            else VALUE_EMPTY_TEXT
+        model.setTextOperations(textOperations)
+        view.setTextOperations(textOperations)
+    }
+
+    fun onEraseButtonPressed() {
+        model.setTextOperations(VALUE_EMPTY_TEXT)
+        view.setTextOperations(VALUE_EMPTY_TEXT)
+    }
+
+    fun onKeyButtonPressed(button: Button) {
+        var textOperations = model.getTextOperations()
+        if (textOperations == MainActivity.OPERATIONS_DEFAULT_TEXT) {
+            textOperations = VALUE_EMPTY_TEXT
+        }
+        textOperations += button.text
+        model.setTextOperations(textOperations)
+        view.setTextOperations(textOperations)
+    }
+
+    fun onEqualButtonPressed() {
+        var textResult = getResults(getItems(model.getTextOperations()))
+        model.setTextResult(textResult)
+        view.setTextResult(textResult)
+    }
+
+    private fun getResults(itemList: List<Key>): String {
+        var textResults = VALUE_EMPTY_TEXT
+        var valeant = mutableListOf<Key>()
+
+        for (vale in itemList) {
+            textResults += "$vale "
+            if (valeant.size < 2) valeant.add(Key(vale.getValue()))
+            else {
+                var aux = VALUE_ZERO
+                when (valeant[1].getValue()) {
+                    "+" -> aux = valeant[0].getInt() + vale.getInt()
+                    "/" -> aux = valeant[0].getInt() / vale.getInt()
+                    "*" -> aux = valeant[0].getInt() * vale.getInt()
+                    "-" -> aux = valeant[0].getInt() - vale.getInt()
+                }
+                textResults += " = $aux\n$aux "
+                valeant = mutableListOf()
+                valeant.add(Key(aux.toString()))
+            }
+        }
+        return textResults
+    }
+
+    private fun getItems(items: CharSequence): List<Key> {
+        var res = mutableListOf<Key>()
+        var num = VALUE_EMPTY_TEXT
+        for (car in items) {
+            if ((car.toInt() < VALUE_MIN_CAR || car.toInt() > VALUE_MAX_CHAR)
+                && car.toInt() != DECIMAL_SEPARATOR.toInt()
+            ) {
+                res.add(Key(num))
+                num = VALUE_EMPTY_TEXT
+                res.add(Key(car.toString()))
+            } else num += car.toString()
+        }
+        if (num != VALUE_EMPTY_TEXT) res.add(Key(num))
+        return res
+    }
+}
+
+/*fun unregister() {
```

Comment:
please remove all the commented code pls

Reply:
Done